extensions/cornerstone/src/tools/ImageOverlayViewerTool.tsx

```diff
@@ -0,0 +1,307 @@
+import { metaData, StackViewport, Types } from '@cornerstonejs/core';
+import { utilities } from '@cornerstonejs/core';
+import { BaseTool } from '@cornerstonejs/tools';
+import { guid } from '@ohif/core/src/utils';
+
+interface CachedStat {
+  color: number[]; // [r, g, b, a]
+  overlays: {
+    // ...overlayPlaneModule
+    _id: string;
+    type: 'G' | 'R'; // G for Graphics, R for ROI
+    color?: number[]; // Rendered color [r, g, b, a]
+    dataUrl?: string; // Rendered image in Data URL expression
+  }[];
+}
+
+const cachedStats: { [key: string]: CachedStat } = {};
+
+/**
+ * compare two RGBA expression of colors.
+ *
+ * @param color1
+ * @param color2
+ * @returns
+ */
+const isSameColor = (color1: number[], color2: number[]) => {
+  return (
+    color1 &&
+    color2 &&
+    color1[0] === color2[0] &&
+    color1[1] === color2[1] &&
+    color1[2] === color2[2] &&
+    color1[3] === color2[3]
+  );
+};
+
+/**
+ * pixelData of overlayPlane module is an array of bits corresponding
+ * to each of the underlying pixels of the image.
+ * Let's create pixel data from bit array of overlay data
+ *
+ * @param pixelDataRaw
+ * @param color
+ * @returns
+ */
+const renderOverlayToDataUrl = ({ width, height }, color, pixelDataRaw) => {
+  const pixelDataView = new DataView(pixelDataRaw);
+  const totalBits = width * height;
+
+  const canvas = document.createElement('canvas');
+  canvas.width = width;
+  canvas.height = height;
+
+  const ctx = canvas.getContext('2d');
+  ctx.clearRect(0, 0, width, height); // make it transparent
+  ctx.globalCompositeOperation = 'copy';
+
+  const imageData = ctx.getImageData(0, 0, width, height);
+  const data = imageData.data;
+  for (let i = 0, bitIdx = 0, byteIdx = 0; i < totalBits; i++) {
+    if (pixelDataView.getUint8(byteIdx) & (1 << bitIdx)) {
+      data[i * 4] = color[0];
+      data[i * 4 + 1] = color[1];
+      data[i * 4 + 2] = color[2];
+      data[i * 4 + 3] = color[3];
+    }
+
+    // next bit, byte
+    if (bitIdx >= 7) {
+      bitIdx = 0;
+      byteIdx++;
+    } else {
+      bitIdx++;
+    }
+  }
+  ctx.putImageData(imageData, 0, 0);
+
+  return canvas.toDataURL();
+};
+
+/**
+ *
+ * @param imageId
+ * @param overlayMetadata
+ * @param color {number[]} - color (r,g,b,a), default color: gray
+ * @returns
+ */
+const getCachedStat = async (
+  imageId: string,
+  overlayMetadata: any[],
+  color: number[] = [127, 127, 127, 255]
+): Promise<CachedStat> => {
+  if (
+    !cachedStats[imageId] ||
+    !isSameColor(cachedStats[imageId].color, color)
+  ) {
```

Comment:
I prefer less indentation. So can we early return on the opposite condition and make indentation one less?